src/styles/block/block.scss

```diff
@@ -0,0 +1,33 @@
+.block {
+  display: grid;
+  grid-template-columns: repeat(3, minmax($box-size, 1fr));
+
+  &__box {
+    text-align: center;
+    line-height: $box-size;
+  }
+  
+  @for $index from 1 through 6 {
+    &__box:nth-child(#{$index})::before {
+      content: "#{$index}";
+    }
+  }
+
+  @each $number, $rgb in $colors {
+    &__box--#{$number} {
```

Comment:
Nice to have different approaches while you are studying, but you could just use only one `block__box` class and use nth-child for coloring it as well